### ISSUE TITLE
some opam packages with C stubs don't build with OCaml 4.03.0+beta1 (No such file or directory)

### ISSUE THREAD

**Issue by edwintorok** [ocaml/ocamlbuild#58]
Trying to install cryptokit or Lwt with OCaml 4.03.0+beta1 fails due to what looks like some path problems.

For example for Lwt 2.5.1:

```
+ /home/edwin/.opam/4.03.0+beta1/bin/ocamlfind ocamlmklib -o src/unix/lwt-unix_stubs -L/usr/lib -lev -L/usr/lib -lpthread src/unix/lwt_unix_stubs.o src/unix/lwt_libev_stubs.o src/unix/lwt_process_stubs.o src/unix/jobs-unix/lwt_unix_job_access.o src/unix/jobs-unix/lwt_unix_job_chdir.o src/unix/jobs-unix/lwt_unix_job_chmod.o src/unix/jobs-unix/lwt_unix_job_chown.o src/unix/jobs-unix/lwt_unix_job_chroot.o src/unix/jobs-unix/lwt_unix_job_close.o src/unix/jobs-unix/lwt_unix_job_fchmod.o src/unix/jobs-unix/lwt_unix_job_fchown.o src/unix/jobs-unix/lwt_unix_job_fdatasync.o src/unix/jobs-unix/lwt_unix_job_fsync.o src/unix/jobs-unix/lwt_unix_job_ftruncate.o src/unix/jobs-unix/lwt_unix_job_link.o src/unix/jobs-unix/lwt_unix_job_lseek.o src/unix/jobs-unix/lwt_unix_job_mkdir.o src/unix/jobs-unix/lwt_unix_job_mkfifo.o src/unix/jobs-unix/lwt_unix_job_rename.o src/unix/jobs-unix/lwt_unix_job_rmdir.o src/unix/jobs-unix/lwt_unix_job_symlink.o src/unix/jobs-unix/lwt_unix_job_tcdrain.o src/unix/jobs-unix/lwt_unix_job_tcflow.o src/unix/jobs-unix/lwt_unix_job_tcflush.o src/unix/jobs-unix/lwt_unix_job_tcsendbreak.o src/unix/jobs-unix/lwt_unix_job_truncate.o src/unix/jobs-unix/lwt_unix_job_unlink.o
gcc: error: src/unix/lwt_unix_stubs.o: No such file or directory
[...]
```

The file `_build/lwt_unix_stubs.o` exists, but `_build/src/unix/lwt_unix_stubs.o` doesn't.
Comparing _build/_log between 4.02.3 and 4.03.0+beta1 shows that 4.02.3 used to do this:

```
# Target: src/unix/lwt_unix_stubs.c, tags: { annot, bin_annot, c, compile, debug, extension:c, file:src/unix/lwt_unix_stubs.c, package(bigarray), package(bytes), package(unix), quiet, safe_string, traverse, use_C_libev, use_C_pthread, use_lwt, use_lwt-log, use_lwt_headers }
/home/edwin/.opam/4.02.3/bin/ocamlfind ocamlc -g -ccopt -I/usr/include -ccopt -I/usr/include -ccopt -Isrc/unix -c src/unix/lwt_unix_stubs.c
# Target: src/unix/lwt_unix_stubs.o, tags: {  }
mv lwt_unix_stubs.o src/unix/lwt_unix_stubs.o
```

While 4.03.0+beta1 does just this (and then the failing mklib call above), I see no `mv` call:

```
# Target: src/unix/lwt_unix_stubs.c, tags: { package(bigarray), package(bytes), package(unix), annot, bin_annot, c, compile, debug, extension:c, file:src/unix/lwt_unix_stubs.c, quiet, safe_string, traverse, use_C_libev, use_C_pthread, use_lwt, use_lwt-log, use_lwt_headers }
/home/edwin/.opam/4.03.0+beta1/bin/ocamlfind ocamlc -g -ccopt -I/usr/include -ccopt -I/usr/include -ccopt -Isrc/unix -package unix -package bytes -package bigarray -c -o src/unix/lwt_unix_stubs.o src/unix/lwt_unix_stubs.c
```

To reproduce:

```
$ opam remote add beta https://github.com/damiendoligez/opam-repository.git#add-compiler-4.03.0+beta1
$ opam switch 4.03.0+beta1
$ eval $(opam config env)
$ opam install lwt
```

What am I doing wrong?

**Comment by gasche** [comment]
Nothing wrong, that's due to a last-minute change in the OCaml-compiler side that is not reflected in the last ocamlbuild release. @bobot remarked that we should do a minor release to fix that (make the last released ocamlbuild version work with the current trunk), I'll do it just now. Sorry for the trouble.

**Comment by whitequark** [comment]
This was fixed in the 0.9.1 release.
